code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright (c) 2020 Aleksej Komarov
+ * SPDX-License-Identifier: MIT
+ */
+
+/proc/tts_filter(text)
+	// Only allow alphanumeric characters and whitespace
+	var/static/regex/bad_chars_regex = regex("\[^a-zA-Z0-9 ,?.!']", "g")
+	return bad_chars_regex.Replace(text, " ")
+
+
+#define TARGET_INDEX 1
+#define IDENTIFIER_INDEX 2
+#define TIMEOUT_INDEX 3
+#define REQUEST_INDEX 4
+
+SUBSYSTEM_DEF(tts)
+	name = "Text To Speech"
+	wait = 0.1 SECONDS
+	init_order = INIT_ORDER_TTS
+
+	var/list/queued_tts_messages = list()
+
+	var/list/processing_tts_messages = list()
+
+	var/list/available_speakers = list()
+
+	var/tts_enabled = FALSE
+
+	var/message_timeout = 5 SECONDS
+
+/datum/controller/subsystem/tts/vv_edit_var(var_name, var_value)
+	// tts being enabled depends on whether it actually exists
+	if(NAMEOF(src, tts_enabled) == var_name)
+		return FALSE
+	return ..()
+
+/datum/controller/subsystem/tts/Initialize()
+	var/datum/http_request/request = new()
+	request.prepare(RUSTG_HTTP_METHOD_GET, "http://localhost:5002/tts-voices", "", "")
```

Comment:
```suggestion
	request.prepare(RUSTG_HTTP_METHOD_GET, "http://192.168.4.91:5002/tts-voices", "", "")
```

i won't be about to do this for a few hours, but this is the ip of our misc docker container host.